// app/tools/iplookup.js
var http = require('http');
var Promise = require('bluebird');

function ipv6to4(ip)
{
	var indexOfColon = ip.lastIndexOf(':');
	var actualIP = ip.substring(indexOfColon+1,ip.length);
	return actualIP;
}

function lookup(ipv4)
{
	return new Promise(function(resolve, reject) {
		var options = {
		    host : 'api.hostip.info',
		    port : 80,
		    path : '/get_json.php',
		    method : 'GET'
		};

		callback = function(response) {
			response.setEncoding('utf8');
			var str = '';

			//another chunk of data has been recieved, so append it to `str`
			response.on('data', function (chunk) {
				str += chunk;
			});

			//the whole response has been recieved, so we just print it out here
			response.on('end', function () {
				var json = JSON.parse(str);
				
				return resolve(json);
			});
		};

		var request = http.request(options, callback);
		request.on('error', function(err) {
		    return reject(err);
		});
		request.end();

	});




	return deferred.promise;
}


module.exports = {
	ipv6to4: ipv6to4,
	lookup: lookup
}
